foxai/explainer/computer_vision/algorithm/xrai.py

```diff
@@ -0,0 +1,687 @@
+
+    Args:
+        array1: First array.
+        array2: Second array.
+
+    Returns:
+        Array of booleans representing result of element-wise AND operation
+            between arrays.
+    """
+    return np.logical_and(array1, np.logical_not(array2))
+
+
+def _get_diff_cnt(array1: np.ndarray, array2: np.ndarray) -> int:
+    """Compute sum of different elements in 2 arrays.
+
+    Args:
+        array1: First array.
+        array2: Second array.
+
+    Returns:
+        Number of different elements between two arrays.
+    """
+    return np.sum(_get_diff_mask(array1, array2))
+
+
+def _unpack_segs_to_masks(segment_list: List[np.ndarray]) -> List[np.ndarray]:
+    """Create masks from segments.
+
+    Args:
+        segment_list: List of segments.
+
+    Returns:
+        List of masks.
+    """
+    masks: List[np.ndarray] = []
+    for segment in segment_list:
+        for index in range(segment.min(), segment.max() + 1):
+            masks.append(segment == index)
+    return masks
+
+
+class XRAI:
+    """A CoreSaliency class that computes saliency masks using the XRAI method."""
+
+    def __init__(self, forward_func: ModelType):
+        # Initialize integrated gradients.
+        self.forward_func = forward_func
+
+    @staticmethod
+    def _get_integrated_gradients(
+        input_data: torch.Tensor,
+        pred_label_idx: TargetType,
+        call_model_function: ModelType,
+        call_model_args: Any,
+        baselines: torch.Tensor,
+        steps: int,
+    ) -> List[AttributionsType]:
+        """Compute Integrated Gradients attributes.
+
+        Args:
+            input_data: Input data.
+            pred_label_idx: Output indices for which gradients are
+                computed (for classification cases, this is usually the
+                target class).
+            call_model_function: The forward function of the model or any
+                modification of it
+            call_model_args: Additional arguments to model forward pass.
+            baselines: Baselines define the starting point from which integral
+                is computed.
+            steps: The number of steps used by the approximation method.
+
+        Returns:
+            List of attributions.
+        """
+        grads: List[AttributionsType] = []
+        integrated_gradients = IntegratedGradientsCVExplainer()
+        for baseline in baselines:
+            grads.append(
+                integrated_gradients.calculate_features(
+                    model=call_model_function,
+                    input_data=input_data,
+                    baselines=baseline,
+                    pred_label_idx=pred_label_idx,
+                    n_steps=steps,
+                    additional_forward_args=call_model_args,
+                )
+            )
+
+        return grads
+
+    @staticmethod
+    def _validate_baselines(
+        input_data: torch.Tensor,
+        baselines: torch.Tensor,
```

Comment:
Is it a BaselineType?

Reply:
Fixed: https://github.com/softwaremill/FoXAI/pull/99/commits/49a269c31fe09c365852daba10e1a93d5fe9850a.